restapi/api_client.go

```diff
@@ -189,7 +189,8 @@ func (client *api_client) send_request(method string, path string, data string)
 		log.Printf("%s\n", body)
 	}
 
-	for num_redirects := client.redirects; num_redirects >= 0; num_redirects-- {
+        /* Retry with redirects and retriable HTTP status codes */
```

Comment:
If I'm not mistaken, I abandoned the redirect logic since it's built-in to the HTTP client so this is dead code. Since this loop pulls it out from practical use, can you also please remove the `num_redirects` variable all around? I ask since you're in the neighborhood :-)
If it's an issue, I can take care of it after merge.